### DISCUSSION TITLE
How to change backend and ui code with hot reloading

### DISCUSSION THREAD

**Discussion by vpeopleonatank** [cvat-ai/cvat#3795] [Q&A] [answered]
When I change some code in cvat (backend) and cvat-ui (ui) folder, then docker images need being rebuilt. Can I enable hot reloading feature when modifying code? Can you share your development workflow?

**Comment by azhavoro** [comment, marked as answer]
https://openvinotoolkit.github.io/cvat/docs/contributing/

**Reply by vpeopleonatank** [reply]
I 've surfed contributing page many times. But ignored the "development environment" section. Thank you!

**Reply by vishalims095** [reply]
> I 've surfed contributing page many times. But ignored the "development environment" section. Thank you!

hello sir can you please help me to execute source code of cvat-ui

**Reply by vpeopleonatank** [reply]
I follow the "development environment" section to install cvat. What problem do you get?

**Reply by vishalims095** [reply]
cvat installed fine and docker image also running fine,
but unable to run source code (npm start/npm build) and .env file also blank
facing issue : 
> webpack-dev-server --env.API_URL=http://localhost:8080 --config ./webpack.config.js --mode=development

**Reply by vpeopleonatank** [reply]
To start cvat-ui, I executed `npm ci && npm run start:cvat-ui` without error.

**Reply by vishalims095** [reply]
hello sir thank you now working fine,  after change in source code cavat-ui  run this command "docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d --build" changes reflects

**Reply by vpeopleonatank** [reply]
Thanks for letting me know.
